ci2/ci/build.py

```diff
@@ -230,19 +262,17 @@ def config(self):
             }
         }]
 
-        sa = None
-        if self.inputs is not None:
-            sa = 'ci2'
-
         self.job = await batch.create_job(CI_UTILS_IMAGE,
                                           command=['bash', '-c', script],
                                           attributes={'name': self.name},
                                           volumes=volumes,
                                           input_files=input_files,
-                                          copy_service_account_name=sa,
                                           parent_ids=self.deps_parent_ids())
 
-    async def cleanup(self, batch, sink):
+    async def cleanup(self, batch, deploy, sink):
+        if deploy and self.publish_as:
```

Comment:
if publish_as is none, then we never pushed the image so we don't need to cleanup, right?

Reply:
Cleanup deletes temporary images (pushed to `gcr.io/<project>/ci-intermediate:<token>`)